analyses/bfa/notebooks/bfa_seas_bavg_iriforecast.md

```diff
@@ -0,0 +1,565 @@
+This notebook explores the option of using IRI's seasonal forecast as the indicator for a drought-related trigger in Burkina Faso. 
+From the country team the proposed trigger is:
+- Trigger #1 in March covering Apr-May-June. Threshold desired: 40%.
+- Trigger #2 in July covering Aug-Sep-Oct. Threshold desired: 50%. 
+- Targeted Admin1s: Boucle de Mounhoun, Centre Nord, Sahel, Nord.
+
+This notebook explores if and when these triggers would be reached. Moreover, an exploration is done on how the raster data can be combined to come to one value for all 4 admin1s.
+
++++
+
+**TODO: look at distribution of normal and above average when high prob below avg**
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+%load_ext autoreload
+%autoreload 2
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+import matplotlib as mpl
+import geopandas as gpd
+from shapely.geometry import mapping
+import pandas as pd
+import rioxarray
+import numpy as np
+import xarray as xr
+import matplotlib.pyplot as plt
+import seaborn as sns
+from rasterstats import zonal_stats
+from IPython.display import Markdown as md
+from myst_nb import glue
+
+from pathlib import Path
+import sys
+import os
+
+path_mod = f"{Path(os.path.dirname(os.path.abspath(''))).parents[1]}/"
+sys.path.append(path_mod)
+from src.indicators.drought.config import Config
+
+from src.indicators.drought.iri_rainfallforecast import get_iri_data
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+adm_sel=["Boucle du Mouhoun","Nord","Centre-Nord","Sahel"]
+threshold_mar=40
+threshold_jul=50
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+country="bfa"
+config=Config()
+parameters = config.parameters(country)
+country_data_raw_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,config.RAW_DIR,country)
+country_data_exploration_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,"exploration",country)
+
+stats_reg_path=os.path.join(country_data_exploration_dir,f"{country}_iri_seasonal_forecast_stats_{''.join(adm_sel)}.csv")
+
+adm1_bound_path=os.path.join(country_data_raw_dir,config.SHAPEFILE_DIR,parameters["path_admin1_shp"])
+adm2_bound_path=os.path.join(country_data_raw_dir,config.SHAPEFILE_DIR,parameters["path_admin2_shp"])
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+iri_ds, iri_transform = get_iri_data(config, download=False)
+```
+
+these are the variables of the forecast data, where C indicates the tercile (below-average, normal, or above-average).  
+F indicates the publication month, and L the leadtime
+
+```{code-cell} ipython3
+iri_ds
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+iri_ds.sel(L=1).prob
+```
+
+```{code-cell} ipython3
+:tags: [remove_cell]
+
+gdf_adm1=gpd.read_file(adm1_bound_path)
+iri_clip=iri_ds.rio.set_spatial_dims(x_dim="lon",y_dim="lat").rio.clip(gdf_adm1.geometry.apply(mapping), iri_ds.rio.crs, all_touched=True)
+```
+
+Below the raw forecast data of below-average rainfall with 1 month leadtime, published in March and July is shown.
```

Comment:
we should specify the period covered by the forecast as well, since multiple forecasts are published at once.

Reply:
With "at once", do you mean with different lead times? As that is what is being selected with L=1 on line 117. So the F which is the publication date, in combination with the L, which is the leadtime, gives the period covered. Does that make sense?